lib/menu.rb

```diff
@@ -0,0 +1,55 @@
+require_relative 'dish'
+
+class Menu
+
+  def initialize(dish_class = Dish)
+    @dish_class = dish_class
+    @list = []
+    load_menu
+  end
+
+  def show
+    print_menu
+  end
+
+  def check(dish)
```

Comment:
Having the `check` method return either a custom symbol like `:unavailable` or the Dish instance makes code that calls this method a bit challenging to reason about since anyone who uses this method will have to remember that different types will be returned depending on the dish.

Rather than having one method do both of:
1. checking if the dish is available on the menu
2. finding the corresponding Dish instance

we could divide these responsibilities into two methods so that:
1. returns a symbol such as `:available`, `:unavailable`, or `:not_on_menu`
2. returns a Dish instance or `nil` if that dish isn't on the menu